frontend/src/components/common/Dropdown/TDropdown.vue

```diff
@@ -0,0 +1,112 @@
+<template>
```

Comment:
There's already more generic component for dropdowns which is relying on slots.
Maybe you can adjust it to fit your case? Or I would suggest to use slot here to avoid creating a dropdown for each kind of the contents.